test/adapters/integration.rb

```diff
@@ -178,6 +178,17 @@ def test_timeout
         end
       end
 
+      def test_timeout_before_open
+        valid_but_inexistent_host = 'http://10.255.255.254'
```

Comment:
We can't affort to maintain external dependencies in our test suite.

The proper way would be to spin up our own test server that's programmed to misbehave, i.e. delay accepting a connection. I'm not sure how to pull that off in Ruby.